contracts/utils/cryptography/MultiSignerERC7913.sol

```diff
@@ -0,0 +1,225 @@
+        _validateReachableThreshold();
+        emit ERC7913ThresholdSet(newThreshold);
+    }
+
+    /// @dev Validates the current threshold is reachable.
+    function _validateReachableThreshold() internal view virtual {
+        uint256 totalSigners = _signers().length();
+        uint256 currentThreshold = threshold();
+        require(
+            totalSigners >= currentThreshold,
+            MultiSignerERC7913UnreachableThreshold(totalSigners, currentThreshold)
+        );
+    }
+
+    /**
+     * @dev Decodes, validates the signature and checks the signers are authorized.
+     * See {_validateNSignatures} and {_validateThreshold} for more details.
+     *
+     * Example of signature encoding:
+     *
+     * ```solidity
+     * // Encode signers (verifier || key)
+     * bytes memory signer1 = abi.encodePacked(verifier1, key1);
+     * bytes memory signer2 = abi.encodePacked(verifier2, key2);
+     *
+     * // Order signers by their id
+     * if (keccak256(signer1) > keccak256(signer2)) {
+     *     (signer1, signer2) = (signer2, signer1);
+     *     (signature1, signature2) = (signature2, signature1);
+     * }
+     *
+     * // Assign ordered signers and signatures
+     * bytes[] memory signers = new bytes[](2);
+     * bytes[] memory signatures = new bytes[](2);
+     * signers[0] = signer1;
+     * signatures[0] = signature1;
+     * signers[1] = signer2;
+     * signatures[1] = signature2;
+     *
+     * // Encode the multi signature
+     * bytes memory signature = abi.encode(signers, signatures);
+     * ```
+     */
+    function _rawSignatureValidation(
+        bytes32 hash,
+        bytes calldata signature
+    ) internal view virtual override returns (bool) {
+        if (signature.length == 0) return false; // For ERC-7739 compatibility
+        (bytes[] memory signingSigners, bytes[] memory signatures) = abi.decode(signature, (bytes[], bytes[]));
+        if (signingSigners.length != signatures.length) return false;
+        return _validateNSignatures(hash, signingSigners, signatures) && _validateThreshold(signingSigners);
+    }
+
+    /**
+     * @dev Validates the signatures using the signers and their corresponding signatures.
+     * Returns whether whether the signers are authorized and the signatures are valid for the given hash.
+     *
+     * IMPORTANT: For simplicity, this contract assumes that the signers are ordered by their {signerId} to
+     * avoid duplication when iterating through the signers (i.e. `signerId(signer1) < signerId(signer2)`).
+     * The function will return false if the signers are not ordered.
+     *
+     * Requirements:
+     *
+     * - The `signers` and `signatures` arrays must be of the same length.
+     */
+    function _validateNSignatures(
+        bytes32 hash,
+        bytes[] memory signingSigners,
+        bytes[] memory signatures
+    ) internal view virtual returns (bool valid) {
+        bytes32 currentSignerId = bytes32(0);
```

Comment:
Previous signer would make more sense here?

Reply:
Addressed in https://github.com/OpenZeppelin/openzeppelin-community-contracts/pull/125/commits/e5bfa3e9905bf0c68b2d51b165a66654b36c28c5